javascripts/discourse/components/kanban/list.gjs

```diff
@@ -0,0 +1,348 @@
+            targetType: "Topic",
+            onSuccess: async () => {
+              this.refreshTopics();
+              oldRefresh();
+            },
+          },
+        });
+      };
+      requireConfirmation = false;
+    } else if (
+      thisDefinition.params.status === "closed" &&
+      oldDefinition.params.status === "open"
+    ) {
+      doUpdate = async () => {
+        await topic.saveStatus("closed", true);
+        this.refreshTopics();
+        oldRefresh();
+      };
+      confirmationMessage = I18n.t(themePrefix("confirm_close"), {
+        title: topic.title,
+      });
+    } else if (
+      thisDefinition.params.status === "open" &&
+      oldDefinition.params.status === "closed"
+    ) {
+      doUpdate = async () => {
+        await topic.saveStatus("closed", false);
+        this.refreshTopics();
+        oldRefresh();
+      };
+      confirmationMessage = I18n.t(themePrefix("confirm_open"), {
+        title: topic.title,
+      });
+    } else if (
+      thisDefinition.params.category &&
+      oldDefinition.params.category &&
+      thisDefinition.params.category !== oldDefinition.params.category
+    ) {
+      doUpdate = async () => {
+        try {
+          await Topic.update(topic, {
+            category_id: thisDefinition.params.category,
+            noBump: true,
+          });
+          this.refreshTopics();
+          oldRefresh();
+        } catch (error) {
+          popupAjaxError(error);
+        }
+      };
+      confirmationMessage = I18n.t(themePrefix("confirm_change_category"), {
+        title: topic.title,
+      });
+    }
+
+    if (requireConfirmation && confirmationMessage) {
+      this.dialog.yesNoConfirm({
+        message: confirmationMessage,
+        didConfirm: doUpdate,
+      });
+    } else {
+      await doUpdate();
+    }
+  }
+
+  @action
+  setDragData(data) {
+    data.oldDefinition = this.args.definition;
+    data.oldRefresh = () => this.refreshTopics();
+    this.args.setDragDataUpstream(data);
+  }
+
+  @action
+  calcHeight(element) {
+    const mainOutlet = element.closest("#main-outlet");
+    const mainOutletHeight = mainOutlet.getBoundingClientRect().height;
+    const mainOutletPadding = 40;
+    const listControlsHeight = mainOutlet
+      .querySelector(".list-controls")
+      .getBoundingClientRect().height;
+    const listTitleHeight = mainOutlet
+      .querySelector(".list-title")
+      .getBoundingClientRect().height;
+    const topicsHeight = document
+      .querySelector(".topics")
+      .getBoundingClientRect().height;
+    const height =
+      mainOutletHeight -
+      listControlsHeight -
+      listTitleHeight -
+      mainOutletPadding;
+
+    element.style.height = `${height}px`;
+  }
+
+  @action
+  addResizeListener(element) {
+    window.addEventListener("resize", () => this.calcHeight(element));
+    this.calcHeight(element);
+  }
+
+  @action
+  removeResizeListener() {
+    window.removeEventListener("resize", this.calcHeight);
+  }
```

Comment:
`addEventListener` is being passed an anonymous function, but `removeEventListener` is being passed `this.calcHeight`.

The result is that `removeEventListener` will be a no-op, and the listener will stick around.

Best bet is probably an inline modifier. Something similar [here](https://github.com/discourse/discourse-kanban-theme/blob/94492f8705743e0c6138250a33191defdbb30560/javascripts/discourse/components/kanban/list.gjs#L32-L44).

e.g.

```js
const onWindowReize = modifier((element, [callback]) => {
  const wrappedCallback = () => callback(element);
  window.addEventListener("resize", wrappedCallback);
  
  return () => { // Cleanup callback
    window.removeEventListener("resize", wrappedCallback);
  }
}
```

And then in the template

```
<div {{onWindowResize this.calcHeight}}>
```